unit_test/async_batcher_test.cc

```diff
@@ -147,16 +167,53 @@ BOOST_AUTO_TEST_CASE(flush_timeout) {
   BOOST_CHECK_EQUAL(result, expected);
 }
 
+BOOST_AUTO_TEST_CASE(flush_timeout_with_counter_ENABLE) {
+  std::vector<std::string> items;
+  std::vector<int> events_count;
+  auto s = new message_sender(items, events_count);
+  size_t timeout_ms = 100; //set a short timeout
+  error_callback_fn error_fn(expect_no_error, nullptr);
+  utility::watchdog watchdog(nullptr);
+  utility::async_batcher_config config;
+  config.send_high_water_mark = 262143;
```

Comment:
Is it a reason to use 262143 here? Also line 180, 8192 is also just a randomly picked number?

Reply:
this is something that was set for already existed test cases. I have used the same parameters but with event_counter_status ENABLE to make sure new implementation is not impacting the existing logic